### PR TITLE
fix: fast fail if no fetched binaries

### PR DESCRIPTION
**Pull request by goldbez**
*Issue #, if available:*

*Description of changes:*
Fast fail if no fetched binaries so that we do not use binary from remote or try to build our own.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -135,6 +135,8 @@ export async function need(opts: NeedOptions) {
     }
   }
 
+  throw new Error("Could not fetch binary.");
```
Comment (sobolk) [inline lib/index.ts:138]:
```suggestion
  throw new Error('Could not find binaries in PKG_CACHE_PATH nor default USER/.pkg-cache. Amplify's fork requires that cache with binaries is prepared before executing this step.');
```

**Review by sobolk — APPROVED** [review]

File: lib/index.ts [hunk 2]
```diff
@@ -135,6 +135,8 @@ export async function need(opts: NeedOptions) {
     }
   }
 
+  throw new Error("Could not find binaries in PKG_CACHE_PATH nor default USER/.pkg-cache. Amplify's fork requires that cache with binaries is prepared before executing this step.");
```
Comment (sdstolworthy) [inline lib/index.ts:138]: Why is this error here rather than after line 134?

Comment (sobolk) [reply lib/index.ts:138]: In case `forceBuild` is set to true. We want to skip logic below in any case.

Comment (goldbez) [reply lib/index.ts:138]: We want it to be outside that if statement so that the error is thrown in the event that the if statement is not entered.

Comment (sdstolworthy) [reply lib/index.ts:138]: I might be misreading the indentation, but doesn't this make line 140 unreachable?

Comment (sobolk) [reply lib/index.ts:138]: Yes, that's the intention.
(plus the fact that we want to keep codebase as close to upstream as possible).

**Review by sdstolworthy — APPROVED** [review]

**Pull request merged**